### DISCUSSION TITLE
What is an exporter?

### DISCUSSION THREAD

**Discussion by fnordian** [grafana-cold-storage/agent#6528] [Q&A]
I've noticed several components in grafana agent that are called `exporter` but they seem to perform different types of activities:

E.g. there is `otelcol.exporter.otlp` that accepts telemetry data from other otelcol components and writes them over the network using the OTLP gRPC protocol.

And there is `prometheus.exporter.snmp` that lets you collect SNMP data and expose them as Prometheus metrics.

So the former is for sending data to an external component, the other one is for gathering data from an external component. What's the intended meaning of `exporter`?

**Comment by rfratto** [comment]
We've been trying to use the appropriate term from both ecosystems. Unfortunately, in this case, Prometheus' term for an exporter doesn't align with OpenTelemetry's term from an exporter. 

In the Prometheus ecosystem, an exporter is a unit of software which connects to a system-under-observation (SUO) and exposes SUO metrics at a Prometheus `/metrics` endpoint. 

In the OpenTelemetry ecosystem, an exporter is a component which sends (exports) telemetry data out. What Prometheus calls an "exporter" would be considered a "receiver" in OpenTelemetry. 

We want our components to feel native to both ecosystems, so I'm not sure we should try to align these terminology, but we can try to improve our documentation to make the difference in terminology between the two ecosystems more clear.
